### PR TITLE
feat: App config restructure

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

This PR Contains the changes of ConfigMap and Secret draft view

Fixes # [AB#4120](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/4120), [AB#4121](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/4121)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Manually tested all the scenarios by creating different different cm and secret

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export default function EnvConfig({ filteredAppIds }: AppGroupDetailDefaultType)
                 <EnvironmentOverride
                     appList={envAppList}
                     environments={environments}
-                    setEnvironments={setEnvironments}
+                    reloadEnvironments={noop}
```
Comment (sohel-rp) [inline src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx:60]: Is noop expected?

File: src/components/app/details/appConfig/AppComposeRouter.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    let index = navItems.findIndex((item) => item.stage === currentStageName)
+    let nextUrl = navItems[index + 1].href
+    if (!isCiPipeline) {
+        return (
+            <div className="app-compose__next-section">
+                <button
+                    type="button"
+                    disabled={isDisabled}
+                    className="cta dc__align-right flex"
+                    onClick={(event) => {
```
Comment (sohel-rp) [inline src/components/app/details/appConfig/AppComposeRouter.tsx:32]: Exiting: Extract to different function

File: src/components/app/details/appConfig/AppComposeRouter.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                            isDisabled={!isUnlocked.workflowEditor}
+                            isCiPipeline={isCiPipeline}
+                        />
+                    </>
+                </Route>
+                {isUnlocked.workflowEditor && [
+                    <Route
+                        key={`${path}/${URLS.APP_WORKFLOW_CONFIG}`}
+                        path={`${path}/${URLS.APP_WORKFLOW_CONFIG}/:workflowId(\\d+)?`}
+                        render={() => (
```
Comment (sohel-rp) [inline src/components/app/details/appConfig/AppComposeRouter.tsx:88]: Existing: use new route syntax

Comment (sohel-rp) [reply src/components/app/details/appConfig/AppComposeRouter.tsx:88]: Same for all routes

File: src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx:60]: yes

**Review by sohel-rp — APPROVED** [review]
To be covered in upcoming PR,
- Remove commented 
- Fix code smells
- Review comments

**Pull request merged**